airbyte-integrations/connectors/source-pokeapi/source_pokeapi/source.py

```diff
@@ -0,0 +1,84 @@
+
+# Basic full refresh stream
+class PokeapiStream(HttpStream):
+    url_base = 'https://pokeapi.co/api/v2/'
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__()
+        self._name = name
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        # The api requires that we include the base currency as a query param so we do that in this method
+        return {"name": self.name}
+
+    def parse_response(
+            self,
+            response: requests.Response,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        # The response is a simple JSON whose schema matches our stream's schema exactly,
+        # so we just return a list containing the response
+        return [response.json()]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # The API does not offer pagination,
+        # so we return None to indicate there are no more pages in the response
+        return None
+
+
+class Pokemon(PokeapiStream):
+    def path(self, **kwargs) -> str:
+        pokemon_name = self._name
+        return f'pokemon/{pokemon_name}'
+
+# Source
+class SourcePokeapi(AbstractSource):
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        input_pokemon = config['name']
+        if input_pokemon not in POKEMON_LIST:
+            return False, f"Input Pokemon {input_pokemon} is invalid. Please check your spelling our input a valid Pokemon."
+        else:
+            return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        args = {"name": config["name"]}
+        return [Pokemon(**args)]
```

Comment:
bundling arguments into a dict like `args` is usually useful when you need to pass those params as a group. In this case we can just pass it directly
```suggestion
        return [Pokemon(name=config["name"])]
```